libs/common/src/main/java/org/opensearch/common/lifecycle/LifecycleComponent.java

```diff
@@ -32,13 +32,15 @@
 
 package org.opensearch.common.lifecycle;
 
+import org.opensearch.common.annotation.PublicApi;
 import org.opensearch.common.lease.Releasable;
 
 /**
  * Base interface for a lifecycle component.
  *
- * @opensearch.internal
+ * @opensearch.api
```

Comment:
@nknize @andrross @dblock to quickly explain why some API (like this one) have been promoted from "wish to be internal" to "it is really a public one":
 - the "root" of the decision is `org.opensearch.plugins.Pluigin`, by definition public API
 - the `org.opensearch.plugins.Pluigin` exposes the following public method (one of many, just an example):
    ```
    public Collection<Class<? extends LifecycleComponent>> getGuiceServiceClasses() {
        return Collections.emptyList();
    }
    ``` 
 - as the consequence, the `org.opensearch.common.lifecycle.LifecycleComponent` is also part of the public API since the plugin developers would have to use it in order to inject own `Guice` service classes

All other changes in this pull request are because of the same API exposure chains.